### PR TITLE
Fix margin positioning for GPG lock icons

### PR DESCRIPTION
**Pull request by CirnoT**
Before:
firefox_2020-06-20_15-43-00
firefox_2020-06-20_15-53-39

After:
firefox_2020-06-20_15-51-02
firefox_2020-06-20_15-54-10

### CODE REVIEW

**Comment by zeripath** [comment]
I think the centerlock needs fixing too - make it a couple px smaller?

**Comment by silverwind** [comment]
> I think the centered cog needs fixing too - make it a couple px smaller?

Imho we should add a custom SVG shape for, those alignment hacks will never work the same everywhere.

**Comment by silverwind** [comment]
Also, should replace the lock with octicons lock (that one isn't fill-styled in v2)

File: templates/repo/shabox_badge.tmpl [hunk 1]
```diff
@@ -2,10 +2,12 @@
 	{{if .verification.Verified}}
 		<div title="{{if eq .verification.TrustStatus "trusted"}}{{else if eq .verification.TrustStatus "untrusted"}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user"}}: {{else}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user_unmatched"}}: {{end}}{{.verification.Reason}}">
 		{{if ne .verification.SigningUser.ID 0}}
-			<i class="lock icon"></i>
+			<i class="icons">
+				<i class="lock icon"></i>
+			</i>
 			<img class="ui signature avatar image" src="{{.verification.SigningUser.RelAvatarLink}}" />
 		{{else}}
-			<i title="{{.verification.Reason}}" class="icons">
```
Comment (silverwind) [inline templates/repo/shabox_badge.tmpl:8]: Why remove the title?

Comment (CirnoT) [reply templates/repo/shabox_badge.tmpl:8]: It's the same as parent div, so it isn't necessary to double it here.

**Comment by CirnoT** [comment]
@zeripath It does but I don't feel like trying to do anything to it at the moment really. Pretty sure we'd need to add negative margin to it.

File: templates/repo/shabox_badge.tmpl (hunk 1 above)
Comment (zeripath) [reply templates/repo/shabox_badge.tmpl:8]: Which div are you talking about?

Comment (CirnoT) [reply templates/repo/shabox_badge.tmpl:8]:
```
<div title="{{if eq .verification.TrustStatus "trusted"}}{{else if eq .verification.TrustStatus "untrusted"}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user"}}: {{else}}{{$.root.i18n.Tr "repo.commits.signed_by_untrusted_user_unmatched"}}: {{end}}{{.verification.Reason}}">
```

Comment (CirnoT) [reply templates/repo/shabox_badge.tmpl:8]: https://github.com/go-gitea/gitea/pull/11995/files/7370a09a184929b31bbef36644b8baff76e92eee#diff-f83349a3876d62b50566b1395e993a0aL3

**Review by zeripath — APPROVED** [review]

**Comment by CirnoT** [comment]
The cog doesn't really work well on either new or old octicon so I've went with what we have currently.

```svg
<svg width="16" height="16" aria-hidden="true" class="svg gitea-lock-cog" viewBox="0 0 14 16">
 <path d="m12.5 7h-0.75v-2.25c0-2.6188-2.1313-4.75-4.75-4.75-2.6187 0-4.75 2.1312-4.75 4.75v2.25h-0.75c-0.82813 0-1.5 0.67188-1.5 1.5v6c0 0.82812 0.67187 1.5 1.5 1.5h11c0.82812 0 1.5-0.67188 1.5-1.5v-6c0-0.82812-0.67188-1.5-1.5-1.5zm-3.25 0h-4.5v-2.25c0-1.2406 1.0094-2.25 2.25-2.25s2.25 1.0094 2.25 2.25z" stroke-width=".03125"/>
 <g transform="translate(.097656)">
  <path d="m8.3471 11.372q0-0.58727-0.41553-1.0028-0.41553-0.41553-1.0028-0.41553-0.58727 0-1.0028 0.41553-0.41553 0.41553-0.41553 1.0028 0 0.58727 0.41553 1.0028 0.41553 0.41553 1.0028 0.41553 0.58727 0 1.0028-0.41553 0.41553-0.41553 0.41553-1.0028zm2.8366-0.60388v1.2299q0 0.06649-0.04435 0.12743-0.04435 0.06092-0.11081 0.07204l-1.0249 0.15512q-0.10526 0.29917-0.21606 0.50415 0.1939 0.27702 0.5928 0.76457 0.05537 0.06649 0.05537 0.13851 0 0.07204-0.04991 0.12743-0.14958 0.20499-0.54847 0.59835-0.39889 0.39336-0.52078 0.39336-0.066485 0-0.14405-0.04981l-0.76454-0.59835q-0.24377 0.12743-0.50415 0.21054-0.088609 0.75346-0.16066 1.0305-0.038796 0.15513-0.19945 0.15513h-1.2299q-0.077594 0-0.13573-0.04703-0.058147-0.04713-0.063703-0.11912l-0.15513-1.0194q-0.27147-0.088609-0.49862-0.20499l-0.78118 0.5928q-0.055368 0.04991-0.13851 0.04991-0.077593 0-0.13851-0.06093-0.69805-0.63159-0.91415-0.93075-0.038797-0.05537-0.038797-0.12743 0-0.06649 0.044353-0.12742 0.08315-0.11634 0.28255-0.36844 0.19945-0.25207 0.29917-0.39057-0.14958-0.27702-0.22714-0.54848l-1.0139-0.14959q-0.072037-0.01111-0.11634-0.06926-0.044353-0.05814-0.044353-0.13019v-1.2299q0-0.06649 0.044353-0.12742 0.044353-0.06092 0.10527-0.07204l1.0305-0.15512q0.077593-0.25485 0.21607-0.5097-0.22161-0.3158-0.5928-0.76454-0.055368-0.066481-0.055368-0.13297 0-0.055368 0.049909-0.12742 0.14404-0.19944 0.54571-0.59556 0.40165-0.39613 0.52355-0.39613 0.072037 0 0.14405 0.055368l0.76454 0.5928q0.24378-0.12743 0.50415-0.21054 0.088609-0.75346 0.16067-1.0305 0.038797-0.15513 0.19944-0.15513h1.2299q0.077594 0 0.13574 0.047131 0.058147 0.047132 0.063703 0.11912l0.15513 1.0194q0.27147 0.088609 0.49862 0.20499l0.7867-0.5928q0.049907-0.049909 0.13297-0.049909 0.07204 0 0.13851 0.055368 0.71469 0.65928 0.91415 0.94184 0.0388 0.044353 0.0388 0.12188 0 0.066481-0.04435 0.12743-0.08315 0.11634-0.28255 0.36841-0.19945 0.25208-0.29917 0.39057 0.14405 0.277 0.22714 0.54295l1.0139 0.15513q0.07204 0.01111 0.11634 0.06926 0.04435 0.05814 0.04435 0.1302z" fill="#fff" fill-rule="evenodd" stroke-width=".031391"/>
 </g>
</svg>
```

```svg
<svg width="16" height="16" aria-hidden="true" class="svg gitea-lock" viewBox="0 0 14 16">
 <path d="m12.5 7h-0.75v-2.25c0-2.6188-2.1313-4.75-4.75-4.75-2.6187 0-4.75 2.1312-4.75 4.75v2.25h-0.75c-0.82813 0-1.5 0.67188-1.5 1.5v6c0 0.82812 0.67187 1.5 1.5 1.5h11c0.82812 0 1.5-0.67188 1.5-1.5v-6c0-0.82812-0.67188-1.5-1.5-1.5zm-3.25 0h-4.5v-2.25c0-1.2406 1.0094-2.25 2.25-2.25s2.25 1.0094 2.25 2.25z" stroke-width=".03125"/>
</svg>
```

```svg
<svg width="16" height="16" aria-hidden="true" class="svg gitea-unlock" viewBox="0 0 14 16">
 <path d="m12.5 8h-7.75v-3.2219c0-1.2375 0.99062-2.2656 2.2281-2.2781 1.25-0.0125 2.2719 1.0031 2.2719 2.25v0.5c0 0.41562 0.33437 0.75 0.75 0.75h1c0.41562 0 0.75-0.33437 0.75-0.75v-0.5c0-2.625-2.1406-4.7594-4.7656-4.75-2.625 0.009375-4.7344 2.1719-4.7344 4.7969v3.2031h-0.75c-0.82812 0-1.5 0.67187-1.5 1.5v5c0 0.82812 0.67187 1.5 1.5 1.5h11c0.82812 0 1.5-0.67187 1.5-1.5v-5c0-0.82812-0.67187-1.5-1.5-1.5z" stroke-width=".03125"/>
</svg>
```

chrome_2020-06-22_02-13-27

**Comment by silverwind** [comment]
Maybe make the cog cutout a tiny bit smaller so there's more of distinct border on top of it?

**Comment by CirnoT** [comment]
It's easy to say to make it smaller when it's at 400%, but the cog already looks almost like a dot when in 100% because the whole icon is 14px by 16px.

**Comment by CirnoT** [comment]
Also I didn't create these SVGs, they're FA icons that we use currently, just in SVG format.

**Comment by CirnoT** [comment]
```svg
<svg class="svg gitea-lock-cog" width="16" height="16" aria-hidden="true" viewBox="0 0 14 16">
 <path d="m12.5 7h-0.75v-2.25c0-2.6188-2.1313-4.75-4.75-4.75s-4.75 2.1312-4.75 4.75v2.25h-0.75c-0.82813 0-1.5 0.67188-1.5 1.5v6c0 0.82812 0.67187 1.5 1.5 1.5h11c0.82812 0 1.5-0.67188 1.5-1.5v-6c0-0.82812-0.67188-1.5-1.5-1.5zm-3.25 0h-4.5v-2.25c0-1.2406 1.0094-2.25 2.25-2.25s2.25 1.0094 2.25 2.25z" stroke-width=".03125"/>
 <g transform="rotate(-90 7.055 11.401)" stroke-width=".2">
  <path d="m8.212 10.779q-0.2371-0.50846-0.76464-0.70047-0.52753-0.19201-1.036 0.045091-0.50846 0.2371-0.70047 0.76464-0.19201 0.52754 0.045095 1.036 0.2371 0.50846 0.76464 0.70047 0.52753 0.19201 1.036-0.04509 0.50846-0.2371 0.70047-0.76464 0.19201-0.52753-0.045095-1.036zm2.2122-1.6681 0.49655 1.0649q0.02684 0.05757 0.01305 0.12824-0.0138 0.07066-0.06685 0.10711l-0.82474 0.54809q0.02965 0.30152 0.01648 0.52373 0.27972 0.16156 0.82194 0.42264 0.07478 0.03521 0.10386 0.09757 0.02909 0.06237 0.0082 0.13048-0.04674 0.23787-0.2333 0.73949-0.18655 0.50162-0.29208 0.55083-0.05757 0.02684-0.14483 0.01503l-0.9035-0.20939q-0.15961 0.20875-0.3515 0.38583 0.22748 0.68813 0.27695 0.95708 0.029042 0.14998-0.11005 0.21484l-1.0649 0.49655q-0.067182 0.03132-0.1365 0.01408-0.069372-0.01733-0.10325-0.07742l-0.54588-0.81998q-0.27082 0.03288-0.51447 0.02383l-0.43702 0.82864q-0.027788 0.06556-0.099773 0.09913-0.067181 0.03133-0.14452 0.0032-0.85937-0.26501-1.1673-0.43678-0.055945-0.03227-0.085039-0.09466-0.026844-0.05757-0.013042-0.12823 0.025022-0.1343 0.095883-0.43308 0.070917-0.29877 0.10134-0.45894-0.24135-0.17946-0.4181-0.38318l-0.93824 0.27983q-0.066856 0.01946-0.12869-0.01299-0.061875-0.03243-0.090963-0.09482l-0.49655-1.0648q-0.026844-0.05757-0.013043-0.12823 0.013806-0.07066 0.062059-0.10487l0.82959-0.5504q-0.035711-0.25198-0.018708-0.52854-0.31937-0.18395-0.82192-0.42261-0.074778-0.035204-0.10162-0.092771-0.022354-0.047938-0.0082348-0.13047 0.044194-0.23084 0.23204-0.73597 0.18782-0.50513 0.29336-0.55435 0.062371-0.029084 0.14707-0.01022l0.90128 0.20458q0.15962-0.20875 0.3515-0.38583-0.22748-0.68813-0.27694-0.95709-0.02904-0.14998 0.11005-0.21483l1.0649-0.49655q0.067182-0.031328 0.13655-0.013996 0.069373 0.017331 0.10325 0.077416l0.54588 0.81998q0.27082-0.032883 0.51447-0.023827l0.4418-0.83087q0.02306-0.06336 0.094977-0.096896 0.062373-0.029085 0.14228-0.00799 0.88496 0.28227 1.1717 0.44638 0.0515 0.022737 0.0828 0.08986 0.026841 0.05756 0.013049 0.12824-0.025021 0.13431-0.095894 0.43305-0.070913 0.29878-0.10134 0.45894 0.23655 0.18167 0.41587 0.37839l0.94048-0.27503q0.06686-0.019466 0.12869 0.012995 0.06188 0.032433 0.09097 0.094823z" fill="#fff" fill-rule="evenodd" stroke-width=".19106"/>
 </g>
</svg>
```

chrome_2020-06-22_12-39-19
inkscape_2020-06-22_12-39-32

**Comment by silverwind** [comment]
Yeah, it's probably just the low dppx that makes them look bad. I'd make sure to run the SVGs throught [SVGOMG](https://jakearchibald.github.io/svgomg/) to optimize them for size.

**Comment by CirnoT** [comment]
```svg
<svg class="svg gitea-lock-cog" width="16" height="16" aria-hidden="true" viewBox="0 0 14 16">
  <path d="M12.5 7h-.75V4.75C11.75 2.131 9.619 0 7 0S2.25 2.131 2.25 4.75V7H1.5A1.5 1.5 0 000 8.5v6A1.5 1.5 0 001.5 16h11a1.5 1.5 0 001.5-1.5v-6A1.5 1.5 0 0012.5 7zM9.25 7h-4.5V4.75C4.75 3.51 5.76 2.5 7 2.5s2.25 1.01 2.25 2.25z"/>
  <path d="M6.433 10.244q-.508.237-.7.765-.192.527.045 1.036.237.508.764.7.528.192 1.036-.045.509-.237.7-.765.193-.527-.044-1.036-.237-.508-.765-.7-.528-.192-1.036.045zM4.765 8.032l1.065-.497q.057-.027.128-.013.07.014.107.067l.548.825q.302-.03.524-.017.162-.28.423-.822.035-.074.097-.103.063-.03.13-.009.239.047.74.234.502.186.551.292.027.057.015.145l-.21.903q.21.16.386.352.689-.228.958-.277.15-.03.214.11l.497 1.065q.031.067.014.136-.017.07-.077.103l-.82.546q.032.271.023.515l.83.437q.065.027.098.1.032.067.003.144-.265.86-.436 1.167-.033.056-.095.085-.058.027-.128.013-.135-.025-.433-.096-.3-.07-.46-.1-.179.24-.382.417l.28.938q.019.067-.014.13-.032.061-.094.09l-1.065.497q-.058.026-.128.013-.071-.014-.105-.062l-.55-.83q-.253.036-.53.019-.183.32-.422.822-.035.074-.093.101-.048.023-.13.009-.231-.045-.736-.233-.505-.187-.554-.293-.03-.062-.01-.147l.204-.901q-.209-.16-.386-.352-.688.228-.957.277-.15.03-.215-.11l-.496-1.065q-.032-.067-.014-.136.017-.07.077-.104l.82-.545q-.033-.271-.024-.515l-.83-.442q-.064-.023-.098-.095-.029-.062-.008-.142.283-.885.447-1.172.023-.051.09-.082.057-.027.128-.014.134.026.433.096.299.071.459.102.182-.237.378-.416l-.275-.94q-.02-.067.013-.13.033-.061.095-.09z" fill="#fff" fill-rule="evenodd" stroke-width=".2"/>
</svg>
```

```svg
<svg width="16" height="16" aria-hidden="true" class="svg gitea-lock" viewBox="0 0 14 16">
  <path d="M12.5 7h-.75V4.75A4.756 4.756 0 007 0a4.756 4.756 0 00-4.75 4.75V7H1.5A1.5 1.5 0 000 8.5v6A1.5 1.5 0 001.5 16h11a1.5 1.5 0 001.5-1.5v-6A1.5 1.5 0 0012.5 7zM9.25 7h-4.5V4.75C4.75 3.51 5.76 2.5 7 2.5s2.25 1.01 2.25 2.25z"/>
</svg>
```

```svg
<svg width="16" height="16" aria-hidden="true" class="svg gitea-unlock" viewBox="0 0 14 16">
  <path d="M12.5 8H4.75V4.778c0-1.237.99-2.265 2.228-2.278A2.253 2.253 0 019.25 4.75v.5c0 .416.334.75.75.75h1c.416 0 .75-.334.75-.75v-.5c0-2.625-2.14-4.76-4.766-4.75C4.36.01 2.25 2.172 2.25 4.797V8H1.5A1.5 1.5 0 000 9.5v5A1.5 1.5 0 001.5 16h11a1.5 1.5 0 001.5-1.5v-5A1.5 1.5 0 0012.5 8z"/>
</svg>
```

**Comment by silverwind** [comment]
If you add this to `webpack.config.json` and put the SVGs in `assets/svg/name.json`, you could probably load them via `{{svg "name" 16}}`.

```diff
diff --git a/webpack.config.js b/webpack.config.js
index 09caeb402..d00c2c74a 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -39,9 +39,12 @@ module.exports = {
     ],
     serviceworker: [
       resolve(__dirname, 'web_src/js/serviceworker.js'),
     ],
-    icons: glob('node_modules/@primer/octicons/build/svg/**/*.svg'),
+    icons: [
+      ...glob('node_modules/@primer/octicons/build/svg/**/*.svg'),
+      ...glob('assets/svg/*.svg'),
+    ],
     ...themes,
   },
   devtool: false,
   output: {

**Comment by CirnoT** [comment]
https://github.com/go-gitea/gitea/pull/12017

**Pull request closed without merging**